static/app/components/profiling/flamegraphSearch.tsx

```diff
@@ -6,9 +6,25 @@ import Fuse from 'fuse.js';
 import space from 'sentry/styles/space';
 import {CanvasPoolManager} from 'sentry/utils/profiling/canvasScheduler';
 import {Flamegraph} from 'sentry/utils/profiling/flamegraph';
+import {useFlamegraphState} from 'sentry/utils/profiling/flamegraph/useFlamegraphState';
 import {FlamegraphFrame} from 'sentry/utils/profiling/flamegraphFrame';
+import {memoizeByReference} from 'sentry/utils/profiling/profile/utils';
 import {isRegExpString, parseRegExp} from 'sentry/utils/profiling/validators/regExp';
 
+function sortFrameResults(
+  frames: Record<string, FlamegraphFrame> | null
+): Array<FlamegraphFrame> {
+  // If frames have the same start times, move frames with lower stack depth first.
+  // This results in top down and left to right iteration
+  return Object.values(frames ?? {}).sort((a, b) =>
+    a.start === b.start
+      ? numericSort(a.depth, b.depth, 'asc')
+      : numericSort(a.start, b.start, 'asc')
+  );
+}
+
+const memoizedSortFrameResults = memoizeByReference(sortFrameResults);
```

Comment:
This is expensive as frames can go to pretty large numbers and we sort on each navigation event. E.g. if you hold up arrow, we will always sort + increment index on key press instead of just increment index.